transcom/mymove
# INT B-18822 Admin API add requested office users get endpoint and updates to office users endpoint

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a907133)

## Summary

To prepare for the backend for the UI to use the functionality of office users requesting accounts, we need to create some additional endpoints to fetch a list of of users from the `office_users` table that are in a `REQUESTED` status.

This PR does the following:
- updates `GET office-users` endpoint to return users in an `APPROVED` status
- updates `POST office-user` endpoint to add the `APPROVED` status to the user on manual creation. We are assuming that is an office user is manually created, they don't need them to be approved
- adds `GET requested-office-users` endpoint that returns a list of `office_users` in a `REQUESTED` status
- updates `swagger-def` to include the endpoint and handle the return values
- updated and added relevant tests

The return value of the new endpoint should include `edipi`, `other_unique_id`, `status`, and `rejection_reason` even if they're null for each object in the array.
```
{
    "active": true,
    "createdAt": "2024-02-27T20:41:07.365Z",
    "edipi": "1231231232",
    "email": "ppm.processing.navy@office.mil",
    "firstName": "Leo",
    "id": "003f8847-942c-488c-b4e5-52375af64ddd",
    "lastName": "Spaceman",
    "middleInitials": null,
    "otherUniqueId": "",
    "rejectionReason": null,
    "roles": [
      {
        "createdAt": "2024-02-27T20:40:41.062Z",
        "id": "010bdae1-8ebe-44c9-b8ee-8c4477fae2a6",
        "roleName": "Services Counselor",
        "roleType": "services_counselor",
        "updatedAt": "2024-02-27T20:40:41.062Z"
      }
    ],
    "status": "REQUESTED",
    "telephone": "415-555-1212",
    "transportationOfficeId": "4de7ec2a-4db7-49a1-ab2b-36945dc23a77",
    "updatedAt": "2024-02-27T20:41:07.365Z",
    "userId": "8bed04b0-9c64-4fb2-bc1a-65223319f109"
  }
```

### How to test

1. Access MM as an admin
2. Manually update some users in the `office_users` table to have `status` of `REQUESTED`
3. Fire up `client` & `server`
4. Log in as an admin
5. Head on over to `http://adminlocal:3000/swagger-ui/admin.html#/`
6. Find the `requested-office-users` endpoint
7. You'll see the added parameters that will allow for pagination when reviewing (mirroring what is being done for viewing admin users)
8. Go ahead and query
9. You should see a return of all the users you changed to `REQUESTED` status in the db and no others

## Screenshots

![Screenshot 2024-02-29 at 11 17 33 AM](https://github.com/transcom/mymove/assets/136510600/92f248b5-36ed-40bb-a9af-fd3ae4dd6650)

![Screenshot 2024-02-29 at 11 24 29 AM](https://github.com/transcom/mymove/assets/136510600/8fcdd69a-1ad4-4ee0-816f-f21e69fcbd92)
